src/components/ModalConfirmCode.tsx

```diff
@@ -1,102 +1,73 @@
-import React,  {FC, useEffect, useState} from 'react';
-import './ModalConfirmCode.scss';
+import React, {useState} from "react";
+import {Link} from "react-router-dom";
+import "./modal/modal.scss";
+import Button from "./Button";
+import Code from "./Code";
+import {Formik, Field, Form} from "formik";
+import {CODE} from "../constants";
 
-interface CardProps {
-  isActive: boolean;
-  setIsActive: (bool: boolean) => void;
+interface FormErrors {
+  [key: string]: string;
 }
 
-const ModalConfirmCode: FC<CardProps> = 
-  ({
-    isActive, 
-    setIsActive,
-  }) => {
-  
-  const [code, setCode] = useState('');
-  const [codeIsDirty, setCodeISDirty] = useState(false);
-  const [codeError, setCodeError] = useState('The input field must be filled');
-  const [isFormValid, setIsFormValid] = useState(false);
-
-  useEffect(() => {
-    setIsFormValid(!codeError);
-  }, [codeError])
+interface FormValue {
+  code: string;
+}
 
-  const sendCode = () => {
-    setIsActive(false);
-  }
+const CodeRegex = new RegExp('^[0-9A-Z]', 'i');
 
-  const blurHandler = () => {
-    setCodeISDirty(true);
+const validate = async (values: FormValue) => {
+  let errors: FormErrors = {};
+  if (!CodeRegex.test(values.code)) {
+    errors.code = 'An invalid character is present in the Code';
```

Comment:
There is a problem here, you store all errors in  errors.code

In case of 2 errors you will have only last record in errors.code at the end.

Interface of this errors object you wrote as

interface FormErrors {
  [key: string]: string;
}